optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,512 @@
+        else:
+            state_dict = self.model.state_dict()
+            torch.save(state_dict, os.path.join(save_directory, file_name))
+        logger.info(f"Model weights saved to {save_directory}")
+
+    @classmethod
+    def _from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str, None]] = None,
+        revision: Optional[Union[str, None]] = None,
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        file_name: Optional[str] = WEIGHTS_NAME,
+        local_files_only: bool = False,
+        use_cache: bool = True,
+        torch_dtype: Optional[Union[str, "torch.dtype"]] = None,
+        **kwargs,
+    ):
+        """
+        Loads a model and its configuration file from a directory or the HF Hub.
+
+        Arguments:
+            model_id (`str` or `Path`):
+                The directory from which to load the model.
+                Can be either:
+                    - The model id of a pretrained model hosted inside a model repo on huggingface.co.
+                    - The path to a directory containing the model weights.
+            use_auth_token (`str` or `bool`):
+                The token to use as HTTP bearer authorization for remote files. Needed to load models from a private
+                repository.
+            revision (`str`, *optional*):
+                The specific model version to use. It can be a branch name, a tag name, or a commit id.
+            cache_dir (`Union[str, Path]`, *optional*):
+                The path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
+            force_download (`bool`, defaults to `False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            file_name(`str`, *optional*):
+                The file name of the model to load. Overwrites the default file name and allows one to load the model
+                with a different name.
+            local_files_only(`bool`, *optional*, defaults to `False`):
+                Whether or not to only look at local files (i.e., do not try to download the model).
+        """
+        backend = getattr(config, "backend", None)
+        if backend == "ipex":
+            pass
+
+        if config.torchscript or backend == "neural_engine":
+            # Load the model from local directory
+            if os.path.isdir(model_id):
+                file_name = os.path.join(model_id, file_name)
+                model_save_dir = model_id
+            # Download the model from the hub
+            else:
+                if backend == "neural_engine":
+                    model_file_names = {"model": ENGINE_MODEL_NAME, "config": ENGINE_MODEL_CONFIG}
```

Comment:
I think we should use `WEIGHTS_NAME` instead, why use `ENGINE_MODEL_NAME` ?
```suggestion
                    model_file_names = {"model": WEIGHTS_NAME, "config": ENGINE_MODEL_CONFIG}
```

Reply:
Now, the neural engine only support fixed model name for engine ir model: model.bin, so I add ENGINE_MODEL_NAME.